models/ignore_and_fire.h

```diff
@@ -0,0 +1,309 @@
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU General Public License for more details.
+ *
+ *  You should have received a copy of the GNU General Public License
+ *  along with NEST.  If not, see <http://www.gnu.org/licenses/>.
+ *
+ */
+
+#ifndef IGNORE_AND_FIRE_H
+#define IGNORE_AND_FIRE_H
+
+// Includes from nestkernel:
+#include "archiving_node.h"
+#include "connection.h"
+#include "event.h"
+#include "nest_types.h"
+#include "ring_buffer.h"
+#include "universal_data_logger.h"
+
+namespace nest
+{
+
+/* BeginUserDocs: neuron, integrate-and-fire, current-based
+
+Short description
++++++++++++++++++
+
+Neuron generating spikes at fixed intervals irrespective of inputs.
+
+Description
++++++++++++
+
+The ``ignore_and_fire`` neuron is a neuron model generating spikes at a predefined ``rate`` with a constant inter-spike
+interval ("fire"), irrespective of its inputs ("ignore"). In this simplest version of the ``ignore_and_fire`` neuron,
+the inputs from other neurons or devices are not processed at all (*).
+
+The model's state variable, the ``phase``, describes the time to the next spike relative to the firing period (the
+inverse of the ``rate``). In each update step, the ``phase`` is decremented by a fixed amount. If it hits zero, a spike
+is emitted and the ``phase`` is reset to +1.
+
+To create asynchronous activity for a population of ``ignore_and_fire`` neurons, the firing phases can be randomly
+initialized.
+
+The ``ignore_and_fire`` neuron is primarily used for neuronal-network model verification and validation purposes
+("benchmarking"), in particular, to evaluate the correctness and performance of connectivity generation and inter-neuron
+communication. It permits an easy scaling of the network size and/or connectivity without affecting the output spike
+statistics. The amount of network traffic is predefined by the user, and therefore fully controllable and predictable,
+irrespective of the network size and structure.
+
+(*) The model can easily be extended and equipped with any arbitrary input processing (such as calculating input
+currents with alpha-function shaped PSC kernels or updating the gating variables in the Hodgkin-Huxley model) or
+(after-) spike generation dynamics to make it more similar and comparable to other non-ignorant neuron models. In such
+extended ignore_and_fire models, the spike emission process would still be decoupled from the intrinsic neuron dynamics.
+
+Parameters
+++++++++++
+
+The following parameters can be set in the status dictionary.
+
+================= ======= ======================================================
+ ``phase``                     Phase (relative time to next spike; 0<phase<=1)
+ ``rate``               1/s    Firing rate
+================= ======= ======================================================
+
+Sends
++++++
+
+SpikeEvent
+
+Receives
+++++++++
+
+SpikeEvent, CurrentEvent, DataLoggingRequest
+
```

Comment:
```suggestion

Examples using this model
+++++++++++++++++++++++++

.. listexamples:: ignore_and_fire

```

I know there are no examples currently for this model, but having this directive here will make sure if we do add examples with ignore_and_fire, they will automatically be listed here.